pkg/scraper/scraper.go

```diff
@@ -137,7 +138,11 @@ func (c *scraper) Scrape(baseCtx context.Context) *storage.MetricsBatch {
 			klog.V(2).InfoS("Scraping node", "node", klog.KObj(node))
 			m, err := c.collectNode(ctx, node)
 			if err != nil {
-				klog.ErrorS(err, "Failed to scrape node", "node", klog.KObj(node))
+				if errors.Is(err, context.DeadlineExceeded) {
```

Comment:
With this change, timeout is no longer based on the metric-resolution. This means that there is no risk of timing out due to sleep. Please remove `-sleepDuration` from line 136 in `ctx, cancelTimeout := context.WithTimeout(baseCtx, c.scrapeTimeout-sleepDuration)`.

Reply:
yeah, I have updated